core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt

```diff
@@ -165,4 +170,273 @@ object GraphQLUtils {
         }
     }
 
+
+    data class GraphInfo(
+            var fields: MutableSet<String> = mutableSetOf(),
+            var edges: MutableSet<String> = mutableSetOf()//set to omit redundant nodes
+
+    )
+
+    data class LongestInfo(
+            var size: Int = 0,
+            var path: List<String> = listOf()
+    )
+
+    fun constructGraph(state: GraphQLActionBuilder.TempState, entryPoint: String, source: String, graph: MutableMap<String, GraphInfo>, history: MutableList<String>, objectFieldsHistory: MutableSet<String>): MutableMap<String, GraphInfo> {
```

Comment:
add comments / javadocs to explain the inputs. eg, in `graph`, what is `String` key?

Reply:
Done.